py/farm_ng/core/events_file_writer.py

```diff
@@ -36,24 +41,32 @@ def open(self) -> bool:
         return self.is_open()
 
     def close(self) -> bool:
-        self._file_stream.close()
+        maybe_file_stream = self._file_stream
+        if maybe_file_stream is None:
+            return False
+        file_stream = cast(IO, maybe_file_stream)
+
+        file_stream.close()
         self._file_stream = None
         return self.is_closed()
 
-    def write(self, message: Any, uri: Optional[event_pb2.Uri] = None) -> None:
+    def write(self, message: Any) -> None:
+        maybe_file_stream = self._file_stream
+        if maybe_file_stream is None:
+            return
+        file_stream = cast(IO, maybe_file_stream)
 
-        if uri is None:
-            uri = event_pb2.Uri()
+        # create a Uri to store the descriptor of the message
+        # it comes in the form of `farm_ng.core.proto.Timestamp`
+        uri = uri_pb2.Uri(scheme=message.DESCRIPTOR.full_name)
```

Comment:
@ethanrublee

Reply:
I made this part of the query, and used scheme="protobuf", query="type=farm_ng.core.proto.Timestamp",